misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt

```diff
@@ -321,11 +321,33 @@ internal class RealTransacter private constructor(
   }
 
   private fun isMessageRetryable(th: SQLException) =
-      // This is thrown as a raw SQLException from Hikari even though it is
-      // most certainly a recoverable exception
-      // See com/zaxxer/hikari/pool/ProxyConnection.java:493
+      isConnectionClosed(th) || isVitessTransactionNotFound(th)
+
+  /**
+   * This is thrown as a raw SQLException from Hikari even though it is most certainly a
+   * recoverable exception.
+   * See com/zaxxer/hikari/pool/ProxyConnection.java:493
+   */
+  private fun isConnectionClosed(th: SQLException) =
       th.message.equals("Connection is closed")
```

Comment:
Could `th.message` be null here, as checked below?

Reply:
Kotlin isn't complaining (otherwise it would be a compile time error) so I think `equals` is defined on `String?` (i.e. nullable String).